### DISCUSSION TITLE
Not able to generate Graphs using GLSP's vscode extension in BluePrint Theia

### DISCUSSION THREAD

**Discussion by sathya-1994** [eclipse-glsp/glsp#486] [Q&A]
Currently we are using GLSP's vscode extension(https://github.com/eclipse-glsp/glsp-vscode-integration) and we have customized few things according our needs.

We tried to install above extension into Blueprint Theia(https://download.eclipse.org/theia/latest/windows/) and extension was installed successfully. But we were not able to generate graphs. When we select .wf file in the explorer, new editor will not be loaded/displayed.

Command to install GLSP's vscode extenion in BluePrint Theia: **TheiaBlueprint.exe start --plugins=local-dir:"C:\glsp-vscode-integration\example\workflow"**

**Our reequipment is to use common extension for Theia and vscode.**

Is there any know issues related to using GLSP's vscode in the Theia?

Steps to reproduce the issue:
 1. Download and Install Blueprint Theia(Or we can build directly by cloning git repo; 2.7.6 version python is needed)

 2. Install GLSP's vscode extension using above command(path needs to be changed)

 3. Try to open .wf file

 4. We can see custom editor contribution for .wf file but custom editor/diagram will not be opened.

**Screenshot:**
![image](https://user-images.githubusercontent.com/59145404/146734090-40e1dc8b-322e-4e12-b5a4-0348155946b0.png)

**Comment by planger** [comment]
Can you please post log files? Otherwise it'll be hard to clearly identify what is going on. Thanks!

**Reply by sathya-1994** [reply]
@planger ,
Attached Blueprint Theia's log file.
[main.log](https://github.com/eclipse-glsp/glsp/files/7744117/main.log)

**Screenshot: Developers tools when we try to open .wf file**
![image](https://user-images.githubusercontent.com/59145404/146744799-ba003dbd-02bd-4c2e-aa8e-804b78f64200.png)

**Reply by planger** [reply]
Hm, could be related to an inconsistency in the versions due to the Codicons (see also https://github.com/eclipse-glsp/glsp/issues/406).
But other than that I don't see anything GLSP specific :/

**Reply by sathya-1994** [reply]
@planger ,

I think issue is not related to GLSP. There might some issue related to loading custom editors in Theia.

I tried with vscode's custom editor API(https://code.visualstudio.com/api/extension-guides/custom-editors) example https://github.com/microsoft/vscode-extension-samples/tree/main/custom-editor-sample in Blueprint Theia. Diagram was not loaded properly for this example also.

**Screenshot:** 
![image](https://user-images.githubusercontent.com/59145404/146781162-af6036bf-d6a4-4259-bf6e-4d603b6efe22.png)

**Main issue:** **Webview will load external files as file URI. If custom editor's webview try to load external file then we are facing net::ERR_ABORTED 404 issue in the Theia. But this is working fine in vscode.**

I found https://github.com/eclipse-theia/theia/issues/9918 ticket where similar issue was reported and I update my findings.
